x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/Mode.java

```diff
@@ -17,10 +17,11 @@
     ODBC;
 
     public static Mode fromString(String mode) {
-        if (mode == null) {
+        try {
+            return Mode.valueOf(mode.toUpperCase(Locale.ROOT));
+        } catch (Exception e) {
```

Comment:
I've done this way because `mode` is a required parameter, even if a client can choose not to send one or send the wrong one. It's just required everywhere in the code.